samples/nhibernate/simple/NHibernate_7/Server/OrderSaga.cs

```diff
@@ -0,0 +1,58 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+
+#region OrderSaga
+
+public class OrderSaga :
+    Saga<Server.OrderSagaData>,
+    IAmStartedByMessages<StartOrder>,
+    IHandleTimeouts<CompleteOrder>
+{
+    static ILog log = LogManager.GetLogger<OrderSaga>();
+
+    protected override void ConfigureHowToFindSaga(SagaPropertyMapper<Server.OrderSagaData> mapper)
+    {
+        mapper.ConfigureMapping<StartOrder>(message => message.OrderId)
+            .ToSaga(sagaData => sagaData.OrderId);
+    }
+
+    public async Task Handle(StartOrder message, IMessageHandlerContext context)
+    {
+        Data.OrderId = message.OrderId;
+        var orderDescription = $"The saga for order {message.OrderId}";
+        Data.OrderDescription = orderDescription;
+        log.Info($"Received StartOrder message {Data.OrderId}. Starting Saga");
+
+        var shipOrder = new ShipOrder
+        {
+            OrderId = message.OrderId
+        };
+        await context.SendLocal(shipOrder)
+            .ConfigureAwait(false);
+
+        log.Info("Order will complete in 5 seconds");
+        var timeoutData = new CompleteOrder
+        {
+            OrderDescription = orderDescription
+        };
+
+        await RequestTimeout(context, TimeSpan.FromSeconds(5), timeoutData)
+            .ConfigureAwait(false);
+    }
+
+    public async Task Timeout(CompleteOrder state, IMessageHandlerContext context)
+    {
+        log.Info($"Saga with OrderId {Data.OrderId} completed");
+        var orderCompleted = new OrderCompleted
+        {
+            OrderId = Data.OrderId
+        };
+        await context.Publish(orderCompleted)
```

Comment:
If we swap these two statements we can use `return context.Publish`. With batching we won't send out the event right away anyway.